spec/features/complaints/tta_activity_report_spec.rb

```diff
@@ -50,6 +43,38 @@
         expect(page).to_not have_content test_display_id
       end
     end
+
+    context "error handling" do
+      before do
+        click_button "Link TTA Activity Report"
+        fill_in "TTA report display ID", with: test_display_id
+        click_button "Link"
+      end
+
+      context "unauthorized user" do
```

Comment:
[question] i am confused as to how these are working/displaying the correct error. can you please explain? thanks!

Reply:
https://github.com/OHS-Hosting-Infrastructure/complaint-tracker/pull/165/files#diff-7292d49acacbcc2165732b0aad4873886ac24363ff4f152ad9f62f70ac5ee701R13 triggers different error conditions based on the `test_display_id`

Reply:
For each of these tests, the context groups the expectation to see a given error message with the `let(:test_display_id)` that will trigger it, and the `before` block actually does the steps to view the final page.

Reply:
oh hmm i see. i did kind of gloss over that while i was reviewing it. so if the 40X error is in the display ID, it will trigger that error message. would you mind adding a comment to the code about what that's doing, and how it helps with development/testing?

Reply:
maybe both? just to keep future me in the loop